contrib/kube-prometheus/jsonnet/kube-prometheus/prometheus-adapter/prometheus-adapter.libsonnet

```diff
@@ -0,0 +1,199 @@
+local k = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
+
+{
+  _config+:: {
+    namespace: 'default',
+
+    versions+:: {
+      prometheusAdapter: 'latest',
+    },
+
+    imageRepos+:: {
+      prometheusAdapter: 'directxman12/k8s-prometheus-adapter-amd64',
+    },
+
+    prometheusAdapter+:: {
+      name: 'prometheus-adapter',
+      labels: { name: $._config.prometheusAdapter.name },
+      config: |||
+        resourceRules:
+          cpu:
+            containerQuery: sum(rate(container_cpu_usage_seconds_total{<<.LabelMatchers>>}[1m])) by (<<.GroupBy>>)
+            nodeQuery: sum(rate(container_cpu_usage_seconds_total{<<.LabelMatchers>>, id='/'}[1m])) by (<<.GroupBy>>)
+            resources:
+              overrides:
+                node:
+                  resource: node
+                namespace:
+                  resource: namespace
+                pod_name:
+                  resource: pod
+            containerLabel: container_name
+          memory:
+            containerQuery: sum(container_memory_working_set_bytes{<<.LabelMatchers>>}) by (<<.GroupBy>>)
+            nodeQuery: sum(container_memory_working_set_bytes{<<.LabelMatchers>>,id='/'}) by (<<.GroupBy>>)
+            resources:
+              overrides:
+                node:
+                  resource: node
+                namespace:
+                  resource: namespace
+                pod_name:
+                  resource: pod
+            containerLabel: container_name
+          window: 1m
+      |||,
+    },
+  },
+
+  prometheusAdapter+:: {
+    apiService:
+      {
+        apiVersion: 'apiregistration.k8s.io/v1beta1',
+        kind: 'APIService',
```

Comment:
What happens if the user doesn't have aggregation enabled? I guess there are quite some deployments out there which don't have aggregation configured.

Reply:
True, so we should make this an optional component that can be added to the manifests, similar to how we do it with Thanos or the anti affinity?

Reply:
Sounds like a great idea! :+1: Some simple "opt-in" mechanism. If we have a precedence already, even better :-)

Reply:
It's been beta since 1.7, and since the data served is a totally different source than let's say etcd, there is no other way. tl;dr if you don't have api aggregation, your cluster won't work properly :slightly_smiling_face: .

Reply:
oki, in that case should we mention this prerequisite in the readme then?